pkg/apis/autoscaling/v1alpha1/metricsproducer.go

```diff
@@ -46,17 +46,44 @@ type PendingCapacitySpec struct {
 	NodeSelector map[string]string `json:"nodeSelector"`
 }
 
-type ScheduledCapacitySpec struct {
-	// NodeSelector specifies a node group. The selector must uniquely identify a set of nodes.
-	NodeSelector map[string]string `json:"nodeSelector"`
+type Schedule struct {
 	// Behaviors may be layered to achieve complex scheduling autoscaling logic
 	Behaviors []ScheduledBehavior `json:"behaviors"`
+	// Defaults to UTC. Users will specify their schedules assuming this is their timezone
+	// +optional
+	Timezone *string `json:"timezone,omitempty"`
+	// A schedule defaults to this value when no behaviors are active
+	DefaultReplicas int32 `json:"defaultReplicas"`
 }
 
 // ScheduledBehavior defines a crontab which sets the metric to a specific replica value on a schedule.
 type ScheduledBehavior struct {
-	Crontab  string `json:"crontab"`
-	Replicas int32  `json:"replicas"`
+	// The value the MetricsProducer will emit when the current time is within start and end
+	Replicas int32       `json:"replicas"`
+	Start    *TimeConfig `json:"start"`
+	End      *TimeConfig `json:"end"`
+}
+
+type TimeConfig struct {
+	// Specify only one of Crontab or Pattern
+	Crontab *string      `json:"crontab,omitempty"`
+	Pattern *TimePattern `json:"pattern,omitempty"`
+}
+
+// TimePattern is a strongly-typed version of crontabs
+type TimePattern struct {
+	// Each field below can be a list of elements separated by commas.
+	Minutes *string `json:"minute,omitempty"`
+	Hours   *string `json:"hour,omitempty"`
+	Days    *string `json:"day,omitempty"`
+	// List of Numbers 1-12 or
```

Comment:
IMO we should limit this with "one clear way". I'd probably lean the fully spelled out January, Februrary, March, etc.. We can validate that. Let's minimize low impact work like supporting fuzzy matching here :P

Reply:
I think I'd rather go with Jan, Feb, Mar format, as that is what Crontabs do as well. I agree with limiting options.

Reply:
As long as there's a single, simple way to do things :)